src/js/actions/tools.js

```diff
@@ -419,9 +431,131 @@ define(function (require, exports) {
      * @private
      * @type {function()}
      */
-    var _toolModalStateChangedHandler;
+    var _toolModalStateChangedHandler,
+        _vectorMaskHandler;
     
     /**
+     * Change the tool's vector mode
+     * This function will both dispatch a vector_state_changed event for the store
+     * and modify the current tool in order to set its properties to mask mode. 
+     * 
+     * @param {boolean} vectorState
+     * @return {Promise}
+     */
+    var changeVectorState = function (vectorState) {
+        var flux = this.flux,
+            toolStore = flux.store("tool"),
+            prevVectorMode = toolStore.getVectorMode(),
+            vectorMode = vectorState,
+            currentTool = toolStore.getCurrentTool(),
+            firstLaunch = false;
+
+        if (prevVectorMode === vectorMode || !currentTool.handleVectorMaskMode) {
+            return Promise.resolve();
+        }
+
+        var appStore = flux.store("application"),
+            currentDocument = appStore.getCurrentDocument(),
+            currentLayers = currentDocument.layers.selected,
+            currentLayer = currentLayers.first(),
+            initPromise,
+            dispatchPromise;
+
+        if (!currentLayer) {
+            return Promise.resolve();
+        }
+        var nonVectorModeLayer = (currentLayer.kind === currentLayer.layerKinds.BACKGROUND ||
+                currentLayer.kind === currentLayer.layerKinds.VECTOR);
```

Comment:
they are layers that we dont want to support vector masks for, so far that includes shape layers and background layers.